src/solvers/refinement/string_constraint_generator_main.cpp

```diff
@@ -247,12 +245,32 @@ string_constraint_generatort::associate_char_array_to_char_pointer(
   array_string_exprt array_sym =
     to_array_string_expr(fresh_symbol(symbol_name, char_array_type));
   auto insert_result =
-    arrays_of_pointers_.insert(std::make_pair(char_pointer, array_sym));
+    arrays_of_pointers.insert(std::make_pair(char_pointer, array_sym));
   array_string_exprt result = to_array_string_expr(insert_result.first->second);
+#if 0
   add_default_axioms(result);
+#endif
   return result;
 }
 
+void array_poolt::insert(
+  const exprt &pointer_expr,
+  array_string_exprt &array_expr)
+{
+  const exprt &length = array_expr.length();
+  if(length == infinity_exprt(length.type()))
+  {
+    auto pair = length_of_array.insert(
```

Comment:
Is it possible that the same array is inserted more than once here? If so should we check that the length we're about to set matches the one already stored?

Reply:
If the same array is added twice then the invariant at the end of this function should fail.